### DISCUSSION TITLE
Wrong query result.

### DISCUSSION THREAD

**Discussion by Saqibs575** [sqlalchemy/sqlalchemy#11732] [Usage Questions] [answered]
Hi,
MyView is a sqlalchemy view class.

```python
query = session.query(MyView)
count = query.count()
record_count = len(query.all())
```
Note that **count** and **record_count** should be same. But it is not. In my case **count = 807** and **record_count=507**

Also, result obtained by executing raw SQL query and sqlalchemy query are different

```python
raw_query_result = session.execute("SELECT * FROM ViewTable").fetchall()
alchemy_result = session.query(MyView).all()
```
**raw_query_result** fetched correct result but **alchemy_result** missed some of the records.

Please help me to fix this

**Comment by gordthompson** [comment]
Have you seen this?

https://docs.sqlalchemy.org/en/20/faq/sessions.html#my-query-does-not-return-the-same-number-of-objects-as-query-count-tells-me-why

**Reply by CaselIT** [reply, marked as answer]
TLDR: use select instead of query

**Comment by Saqibs575** [comment]
Thanks @gordthompson and @CaselIT
